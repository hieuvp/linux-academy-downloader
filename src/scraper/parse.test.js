const fs = require('fs');

const parse = require('./parse');


describe('parse', () => {
  it('should return', () => {
    const html = fs.readFileSync(
      `${__dirname}/test-data/course-mastering-systemd.html`,
      'utf8',
    );
    expect(parse(html)).toMatchSnapshot();
  });
});
