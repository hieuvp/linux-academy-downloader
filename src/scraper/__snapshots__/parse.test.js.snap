// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse should return 1`] = `
Array [
  Object {
    "course": "Mastering Systemd",
    "lesson": "Course Introduction",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1710/lesson/1",
    "section": "Course Introduction",
    "subsection": "Getting Started",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "About the Course Author",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1710/lesson/2",
    "section": "Course Introduction",
    "subsection": "Getting Started",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "The Linux Boot Process",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1711/lesson/1",
    "section": "Linux Daemon History",
    "subsection": "Previous Service Management Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "init",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1711/lesson/2",
    "section": "Linux Daemon History",
    "subsection": "Previous Service Management Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Red Hat Service Tools",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1711/lesson/3",
    "section": "Linux Daemon History",
    "subsection": "Previous Service Management Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "upstart",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1711/lesson/4",
    "section": "Linux Daemon History",
    "subsection": "Previous Service Management Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "The Purpose of systemd",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1712/lesson/1",
    "section": "Linux Daemon History",
    "subsection": "systemd Introduction",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "systemd Architecture",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1712/lesson/2",
    "section": "Linux Daemon History",
    "subsection": "systemd Introduction",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Alternatives to systemd",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1712/lesson/3",
    "section": "Linux Daemon History",
    "subsection": "systemd Introduction",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "QUIZ: Linux Daemon History",
    "link": "https://app.linuxacademy.com/challenges/eef1813f-e6bf-4ff6-9892-16416855d09a?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "Linux Daemon History",
    "subsection": "systemd Introduction",
    "type": "quiz",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "systemctl",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1713/lesson/1",
    "section": "systemd Components",
    "subsection": "systemd Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Introduction to the systemd Journal",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1713/lesson/2",
    "section": "systemd Components",
    "subsection": "systemd Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "journalctl",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1713/lesson/3",
    "section": "systemd Components",
    "subsection": "systemd Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "More systemd Tools",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1713/lesson/4",
    "section": "systemd Components",
    "subsection": "systemd Tools",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "QUIZ: Systemd Components",
    "link": "https://app.linuxacademy.com/challenges/492d694e-56cd-4124-9182-49d1d64c6ccc?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "systemd Components",
    "subsection": "systemd Tools",
    "type": "quiz",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Basics of a Unit File",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1714/lesson/1",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Target Unit Files",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1714/lesson/2",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Service Unit Files",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1714/lesson/3",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Timer Unit Files",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1714/lesson/4",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Working with Services in Systemd - Start/Stop/Restart/Status",
    "link": "https://app.linuxacademy.com/hands-on-labs/5ac5711e-15ac-441b-bf5e-d8df6742063e?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "lab",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Changing the Default Target of a System",
    "link": "https://app.linuxacademy.com/hands-on-labs/585811f4-290e-4c6c-8c12-030d64be9352?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "lab",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "QUIZ: Systemd Unit Files",
    "link": "https://app.linuxacademy.com/challenges/e18a441c-c617-4469-8425-ad88314486f7?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "systemd Components",
    "subsection": "Unit Files",
    "type": "quiz",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "systemd and Containers",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1888/lesson/1",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "systemd Container Demo",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1888/lesson/2",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Networking in systemd Containers",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1888/lesson/3",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "video",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Creating a Systemd Compatible CentOS 7 Container",
    "link": "https://app.linuxacademy.com/hands-on-labs/876138d8-37e2-4755-81e6-1ca98f884b6d?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "lab",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Creating a Systemd Compatible Debian Container ",
    "link": "https://app.linuxacademy.com/hands-on-labs/1eca0d56-b853-4ebb-8917-d7a0a3018fd5?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "lab",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "QUIZ: Systemd and Containers",
    "link": "https://app.linuxacademy.com/challenges/3c3101cf-d2bb-48ad-810e-608e91a52ea8?redirect_uri=https://linuxacademy.com/cp/modules/view/id/171",
    "section": "Containers with systemd",
    "subsection": "systemd and Containers",
    "type": "quiz",
  },
  Object {
    "course": "Mastering Systemd",
    "lesson": "Final Steps",
    "link": "https://linuxacademy.com/cp/courses/lesson/course/1715/lesson/1",
    "section": "Course Conclusion",
    "subsection": "Final Steps",
    "type": "video",
  },
]
`;
